src/math/Math.sol

```diff
@@ -28,4 +28,16 @@ library Math {
             }
         }
     }
+
+    /// @dev Returns x / y rounded up.
+    function divUp(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        assembly {
+            // Revert if a + b < a or a + b < b
```

Comment:
```suggestion
            // Overflows if a + b < a or a + b < b
```

Reply:
and thus it explains why we revert (I'd move this comment down a line ideally)

Reply:
The only way a + b < a or b is if  a + b overflowed, right? In that case, we should only need to do one check against a or b. and then we can change the comment to something like:
// Reverts if a + b overflows

Reply:
Proof:

Let n = 2^256. We want to check that a + b < n for overflow.
Let x = (a + b) % n
a < n & b < n

If a + b < n, then x = a + b
x >= a

If a + b >= n, then x = a + b - n
b < n so
x < a + n - n = a

This math works for a or b.

Reply:
Good point @pakim249CAL

Reply:
agree with you too @Rubilmax

Reply:
Tried @ewile version but this one seems to be better.